plugins/inputs/systemd_units/systemd_units_linux.go

```diff
@@ -158,12 +161,34 @@ func (s *SystemdUnits) Init() error {
 	}
 	s.filter = f
 
+	switch s.Scope {
+	case "", "system":
+		s.scope = "system"
+		s.user = ""
+	case "user":
+		s.scope = "user"
+		s.user = os.Getenv("USER")
+	default:
+		return fmt.Errorf("invalid 'scope' %q", s.Scope)
+	}
```

Comment:
Hmmm I don't like the user to be passed in as an environment variable as this scatters the information. It would be better to determine the user e.g. using [user.Current()](https://pkg.go.dev/os/user#Current) here. What do you think?

Reply:
Good idea! I modified the code to use os/user instead.